runtime/jvmti/jvmtiExtensionMechanism.c

```diff
@@ -3683,3 +3734,131 @@ jvmtiDeregisterTracePointSubscriber(jvmtiEnv* env, ...)
 	done:
 	TRACE_JVMTI_RETURN(jvmtiDeregisterTracePointSubscriber);
 }
+
+#if JAVA_SPEC_VERSION >= 19
+/**
+ * Get the virtual thread corresponding to a carrier thread.
+ *
+ * @param[in] jvmti_env the jvmti env.
+ * @param[in] carrier_thread the carrier thread.
+ * @param[out] virtual_thread_ptr the pointer to the virtual threaed.
+ * @return JVMTI_ERROR_NONE on success, or a JVMTI error on failure.
+ */
+static jvmtiError JNICALL
+jvmtiGetVirtualThread(jvmtiEnv* jvmti_env, ...)
+{
+	J9JavaVM *vm = JAVAVM_FROM_ENV(jvmti_env);
+	jvmtiError rc = JVMTI_ERROR_NONE;
+	J9VMThread *currentThread = NULL;
+	jthread rv_virtual_thread = NULL;
+
+	jthread carrier_thread = NULL;
+	jthread *virtual_thread_ptr = NULL;
+	va_list args;
+	va_start(args, jvmti_env);
+	carrier_thread = va_arg(args, jthread);
+	virtual_thread_ptr = va_arg(args, jthread*);
+	va_end(args);
+
+	Trc_JVMTI_jvmtiGetVirtualThread_Entry(jvmti_env);
+
+	rc = getCurrentVMThread(vm, &currentThread);
+	if (rc == JVMTI_ERROR_NONE) {
```

Comment:
Invert the test - I realize this is copied code but we may as well fix the new instances. Another one below.

Reply:
Inverted for all instances.